common/lib/main.s

```diff
@@ -0,0 +1,20 @@
+.text
+.global __main
+.global __exit
+
+__main:
+    TSX
+    STX __save_sp
+    JMP main
+
+__exit:
+   STX __save_x
+   LDX __save_sp
+   TXS
+   LDX __save_x
+   rts
+
+__save_sp:
```

Comment:
__save_s; we've taking to calling the hardware stack register S and the soft stack pointer SP.